src/reducers/url-state.js

```diff
@@ -284,8 +286,36 @@ export const getHidePlatformDetails = (state: State) =>
   getUrlState(state).hidePlatformDetails;
 export const getInvertCallstack = (state: State) =>
   getUrlState(state).invertCallstack;
-export const getSearchString = (state: State) =>
+export const getCurrentSearchString = (state: State) =>
   getUrlState(state).callTreeSearchString;
+export const getSearchStrings = createSelector(
+  getCurrentSearchString,
+  searchString => {
+    if (!searchString) {
+      return null;
+    }
+    const result = searchString
+      .split(',')
+      .map(part => part.trim())
+      .filter(part => part);
+
+    if (result.length) {
+      return result;
+    }
+
+    return null;
+  }
+);
+export const getSearchStringsAsRegExp = createSelector(
+  getSearchStrings,
+  strings => {
+    if (!strings || !strings.length) {
+      return null;
+    }
+    const regexpStr = strings.map(escapeStringRegexp).join('|');
+    return new RegExp(regexpStr, 'gi');
+  }
+);
```

Comment:
This is new ^.

Now we memoize the regexp. But because it's a global regexp we need to take care to reset its lastIndex property when we use it. I've left comments about that everywhere I use it.

Reply:
Thanks!